### PR TITLE
Allow empty host port in --publish flag

### LINKED ISSUE #6806: Publish does not work after upgrade
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

It seems like something changed in the --publish syntax ?

> **--publish**, **-p**=_ip_:_hostPort_:_containerPort_ | _ip_::_containerPort_ | _hostPort_:_containerPort_ | _containerPort_
>
> Publish a container's port, or range of ports, to the host.
>
> Both hostPort and containerPort can be specified as a range of ports.
>
> When specifying ranges for both, the number of container ports in the range must match the number of host ports in the range.

It no longer seems to accept the _ip_::_containerPort_ variant.

**Steps to reproduce the issue:**

1. Use "podman run" with `--publish` parameteres

2.

3.

**Describe the results you received:**

$ sudo podman2 run --publish=127.0.0.1::8443 --publish=127.0.0.1::22 --publish=127.0.0.1::2376 --publish=127.0.0.1::5000 ubuntu:20.04
Error: must provide a non-empty container host port to publish

**Describe the results you expected:**

$ sudo podman1 run --publish=127.0.0.1::8443 --publish=127.0.0.1::22 --publish=127.0.0.1::2376 --publish=127.0.0.1::5000 ubuntu:20.04

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

1.8.2

2.0.0

**Output of `podman info --debug`:**

```
(paste your output here)
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Ubuntu 20.04

### PR DESCRIPTION
**Pull request by mheon**
I didn't believe that this was actually legal, but it looks like it is. Simple enough fix (just remove the error message), too.

Fixes #6806

### CODE REVIEW

File: pkg/specgen/generate/ports.go [hunk 1]
```diff
@@ -105,26 +104,44 @@ func parsePortMapping(portMappings []specgen.PortMapping) ([]ocicni.PortMapping,
...
-				testHPort := hostPortMap[hPort]
-				if testHPort != 0 && testHPort != cPort {
-					return nil, nil, nil, errors.Errorf("conflicting port mappings for host port %d (protocol %s)", hPort, p)
-				}
-				hostPortMap[hPort] = cPort
-
-				// If we have an exact duplicate, just continue
-				if testCPort == hPort && testHPort == cPort {
-					continue
+				if hPort != 0 {
```
Comment (rhatdan) [inline pkg/specgen/generate/ports.go:127]: Why isn't this just an else?

**Comment by rhatdan** [comment]
This should be better documented in the man pages.

File: pkg/specgen/generate/ports.go (hunk 1 above)
Comment (mheon) [reply pkg/specgen/generate/ports.go:127]: Good point

**Comment by mheon** [comment]
Added to manpages

**Comment by mheon** [comment]
Tests are green
@rhatdan @baude @vrothberg @TomSweeneyRedHat @giuseppe @jwhonce PTAL

**Comment by rhatdan** [comment]
LGTM
@baude @TomSweeneyRedHat @vrothberg @giuseppe @ashley-cui @jwhonce @QiWang19 PTAL

**Comment by QiWang19** [comment]
LGTM

**Pull request merged**